File: frontend/app/types/interview.ts
export type InterviewType = 'ONLINE' | 'ONSITE' | 'PHONE';

export interface Interview {
  id: string;
  jobId: string;
  companyName: string;
  positionTitle: string;
  dateTime: string;
  type: InterviewType;
  notes?: string;
  preparationDetails?: string;
}

export interface InterviewFormData {
  jobId: string;
  companyName: string;
  positionTitle: string;
  dateTime: string;
  type: InterviewType;
  notes?: string;
  preparationDetails?: string;
} 

File: Backend/interview_service/src/main/java/com/jobtracker/interview_service/entities/Interview.java
package com.jobtracker.interview_service.entities;

import java.time.LocalDateTime;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "interviews")
public class Interview {

    @Id
    private String id;
    private String jobId;
    private String companyName;
    private String positionTitle;
    private String dateTime;
    private String type; // ONLINE, ONSITE, PHONE
    private String notes;
    private String preparationDetails;
    @CreatedDate
    private LocalDateTime createdDate;
    @LastModifiedDate
    private LocalDateTime lastModifiedDate;
}
